packages/core/src/components/BaseInput/BaseInput.styles.tsx

```diff
@@ -0,0 +1,257 @@
+      },
+      "input[type=search]::-webkit-search-results-decoration": {
+        display: "none",
+      },
+    },
+    ...($resizable && {
+      width: "auto",
+    }),
+    ...($readOnly && {
+      "& .inputBorderContainer": {
+        backgroundColor: "transparent",
+      },
+      "&:hover .inputBorderContainer": {
+        backgroundColor: "transparent",
+      },
+      "&:focus-within .inputBorderContainer": {
+        backgroundColor: "transparent",
+      },
+
+      // multiline
+      "& .inputRootMultiline": {
+        "& .input": {
+          border: "1px solid transparent",
+        },
+      },
+      "&:hover .inputRootMultiline": {
+        "& .input": {
+          border: "1px solid transparent",
+        },
+      },
+      "&:focus-within .inputRootMultiline": {
+        "& .input": {
+          border: "1px solid transparent",
+        },
+      },
+    }),
+    ...($disabled && {
+      "& .inputBorderContainer": {
+        backgroundColor: theme.colors.atmo4,
+      },
+      "&:hover .inputBorderContainer": {
+        backgroundColor: theme.colors.atmo4,
+      },
+      "& .inputRootMultiline": {
+        "& .input": {
+          backgroundColor: theme.colors.atmo3,
+          border: `1px solid ${theme.colors.atmo4}`,
+        },
+      },
+      "&:hover .inputRootMultiline": {
+        "& .input": {
+          backgroundColor: theme.colors.atmo3,
+          border: `1px solid ${theme.colors.atmo4}`,
+        },
+      },
+    }),
+    ...($invalid && {
+      "&:not(.disabled)": {
+        "& .inputBorderContainer": {
+          backgroundColor: theme.colors.sema4,
+        },
+        "&:hover .inputBorderContainer": {
+          backgroundColor: theme.colors.sema4,
+        },
+        "& .inputRootMultiline": {
+          "& .input": {
+            border: `1px solid ${theme.colors.sema4}`,
+          },
+        },
+        "&:hover .inputRootMultiline": {
+          "& .input": {
+            border: `1px solid ${theme.colors.sema4}`,
+          },
+        },
+        "&:focus-within .inputRootMultiline": {
+          "& .input": {
+            border: `1px solid ${theme.colors.sema4}`,
+          },
+        },
+      },
+    }),
+  })
+);
+
+export const StyledInputBorderContainer = styled("div")({
+  position: "absolute",
+  width: "calc(100% - 4px)",
+  height: theme.baseInput.underlineHeight,
+  top: "31px",
+  left: "2px",
+  backgroundColor: theme.colors.atmo4,
+});
+
+export const StyledInput = styled(
+  MuiInput,
+  transientOptions
+)(({ $resizable, $disabled }: { $resizable: boolean; $disabled: boolean }) => ({
+  [`&.${MuiInputClasses.root}`]: {
+    margin: 0,
+    width: "100%",
+    borderRadius: "2px",
+    border: `1px solid ${theme.baseInput.borderColor}`,
+    backgroundColor: theme.colors.atmo1,
+    "&.inputRootInvalid": {
+      border: `1px solid ${theme.baseInput.invalidColor}`,
+    },
+    "&.inputRootReadOnly": {
+      border: `1px solid ${theme.colors.atmo5}`,
+    },
```

Comment:
There's a border around the input when it's readonly but it's only supposed to appear in DS5 and not DS3. Furthermore, in [DS5](https://designsystem.hitachivantara.com/6c705d900/v/30842/p/6700f7-input/b/50959e/i/183325454) it seems like the input has the same `atmo2` background color as the disabled state when it's readonly.

Reply:
Yep, I was working on it. PR updated.